src/main/java/eightbit/task/Deadline.java

```diff
@@ -0,0 +1,48 @@
+package eightbit.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a task with a deadline.
+ */
+public class Deadline extends Task {
+
+    protected LocalDateTime by;
+
+    /**
+     * Constructs a task with a deadline.
+     *
+     * @param description Description of the task.
+     * @param by Date and time of the deadline.
+     */
+    public Deadline(String description, LocalDateTime by) {
+        super(description);
+        this.by = by;
+    }
+
+    /**
+     * Constructs a task with a deadline. Marks the task as done or not.
+     *
+     * @param description Description of the task.
+     * @param by Date and time of the deadline.
+     * @param isDone Whether the task is done.
+     */
+    public Deadline(String description, LocalDateTime by, boolean isDone) {
+        super(description, isDone);
+        this.by = by;
+    }
+
+    public LocalDateTime getBy() {
+        return this.by;
+    }
+
+    /**
+     * @return String representation of a Deadline.
+     */
+    @Override
+    public String toString() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd MMM yyyy HH:mm");
+        return "[D]" + super.toString() + " (by: " + by.format(formatter) + ")";
```

Comment:
In this case, should we use `String.format()` instead, since for me, it's more readable and less verbose?